graph/depth_first_search_with_stack.cpp

```diff
@@ -1,46 +1,208 @@
+/**
+ *
+ * @file
+ * @brief [Depth First Search Algorithm using Stack
+ * (Depth First Search Algorithm)](https://en.wikipedia.org/wiki/Depth-first_search)
+ *
+ * @author [Ayaan Khan](http://github.com/ayaankhan98)
+ * @author [Saurav Uppoor](https://github.com/sauravUppoor)
+ *
+ * @details
+ * Depth First Search also quoted as DFS is a Graph Traversal Algorithm.
+ * Time Complexity O(|V| + |E|) where V is number of vertices and E
+ * is number of edges in graph.
+ *
+ * Application of Depth First Search are
+ *
+ * 1. Finding connected components
+ * 2. Finding 2-(edge or vertex)-connected components.
+ * 3. Finding 3-(edge or vertex)-connected components.
+ * 4. Finding the bridges of a graph.
+ * 5. Generating words in order to plot the limit set of a group.
+ * 6. Finding strongly connected components.
+ *
+ * <h4>Working</h4>
+ * 1. Mark all vertices as unvisited (colour it WHITE).
+ * 2. Push starting vertex into the stack and colour it GREY.
+ * 3. Once a node is popped out of the stack and is coloured GREY, we colour it BLACK.
+ * 4. Push all its neighbours which are not coloured BLACK.
+ * 5. Repeat steps 4 and 5 until the stack is empty.
+ */
+
 #include <iostream>
 #include <list>
 #include <stack>
 #include <vector>
+#include <cassert>
+#include <limits>
```

Comment:
Add a brief description of what the header/library add (see the examples below).
```suggestion
#include <iostream>  /// for io operations
#include <list>     ///
#include <stack>    ///
#include <vector>  /// for std::vector
#include <cassert>  /// for assert
#include <limits>   ///
```